modules/core/src/main/java/org/apache/synapse/util/CommentListUtil.java

```diff
@@ -58,10 +58,12 @@ public static void populateComments(OMElement el, List<String> commentList) {
      */
     public static void serializeComments(OMElement parent, List<String> commentList) {
         OMFactory fac = OMAbstractFactory.getOMFactory();
-        for (String comment : commentList) {
-            OMComment commendNode = fac.createOMComment(parent, "comment");
-            commendNode.setValue(comment);
-            parent.addChild(commendNode);
+        if (commentList != null) {
```

Comment:
Why do we get commentList as null here? Shouldn't we pass an empty list when we do not have any items.

Reply:
Updated.